### DISCUSSION TITLE
How to pass environment variable to post-upgrade task?

### DISCUSSION THREAD

**Discussion by knutwannheden** [renovatebot/renovate#18712] [Request Help] [answered]
I am trying to run a post-upgrade task (in a self-hosted Renovate setup on GitLab) while passing some environment variables to it. But so far I have not succeeded.

My post-upgrade task only gets executed when my config **doesn't** contain any settings like `customEnvVariables`, `secrets` or `exposeAllEnv`. I also tried setting `trustLevel` to `high` in my `config.js`, but that didn't make any difference either.

Since my post-upgrade task **does** get executed when the mentioned options are omitted, it seems like my setup is otherwise sane. Any idea what I am missing?

**Comment by viceice** [comment, marked as answer]
please provide your sanitized config.js

**Reply by knutwannheden** [reply]
I think I may have figured out the "problem": It appears that the post-upgrade tasks only get executed when Renovate updated any of the dependencies. I didn't see anything in the logs that the post-upgrade tasks got skipped for this reason. Instead I was assuming I was either using some illegal combination of settings or was missing some config.

I can now however confirm that everything works as expected, when Renovate actually changes something. While the documentation does state:
> Post-upgrade tasks are commands that are executed by Renovate after a dependency has been updated but before the commit is created.

I think it could make sense to also clarify that the tasks **don't** get executed when nothing got changed. Also a hint in the logs could be nice.

The reason I ran into this is that I am testing my post-upgrade tasks and just rerun Renovate. It finds the GitLab MR it created in the previous run and updates it, as necessary.
